### PR TITLE
Add Serialbox Comparison Script, Update README 

### PR DESCRIPTION
**Pull request by robgpita**
## Proposed changes

Add a Python Script (and supporting code) that uses Serialbox data from Fortran implementation to verify results.

Update README for Python Implementation and Comparison instructions. Create new `shallow_water_serialize`  Fortran program to output Serialbox data, update CMakeLists.txt to link and build new executable.

To test the proposed changes, follow instructions in the updated README under Comparison Tests. 

## Types of changes

What types of changes does your code introduce?

_Replace the space with an `x` for ALL the boxes that apply_

- [ ] Bugfix (Change which fixes an issue)
- [x] New feature (Change which adds functionality)
- [x] Enhancement to existing functionality (non-breaking change which improves existing code or documentation)
- [ ] Breaking change (fix or feature that changes results and requires updates to test baselines)
- [x] Documentation Update

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/NOAA-GSL/SENA-shallow-water/blob/develop/CONTRIBUTING.md) document
- [x] I have ensured all my changes contribute to a common purpose
- [x] I have verified my change does not add debug code
- [x] I have verified my change does not add code that is commented out
- [x] I have verified unit tests pass locally with my changes
- [x] I have verified my changes adhere to style guidelines
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] I have added necessary documentation (if appropriate)

---

## Reviewers' checklist

The following is a checklist for reviewers to consider when reviewing this pull request.

#### Implementation

- [ ] Does this code do what it is supposed to do?
- [ ] Can the code be simplified?
- [ ] Does the code add unwanted compile-time or run-time dependencies?
- [ ] Is the code at the correct abstraction level?
- [ ] Is the code modular enough?
- [ ] Does similar functionality already exist in the codebase? If so, why wasn't it used?
- [ ] Can you think of a different solution that is substantially better in terms of code maintenance, readability, or performance?

#### Logic Errors and Bugs

- [ ] Can you think of a use case in which the code does not behave properly?
- [ ] Can you think of any inputs or external events that could break the code?

#### Error handling and logging

- [ ] Is error handling done correctly?
- [ ] Should any logging or debuggging be added or removed?
- [ ] Are error messages user friendly?

#### Usability

- [ ] Is the API intuitive to use?
- [ ] Is the API well documented?

#### Testing and testability

- [ ] Do the tests pass?
- [ ] Is the code testable?
- [ ] Do the tests reasonably cover the code change?
- [ ] Are there additional edge cases or inputs that should be tested?

#### Performance

- [ ] Do you think the code change will negatively impact performance?
- [ ] Do you see any potential to improve the performance of the code?

#### Dependencies

- [ ] Does this change require updates to documenation, configuration?  If so, what that done?
- [ ] Does this change affect backward compatibility?

#### Readability

- [ ] Is the code easy to understand?
- [ ] Can the readability be improved with smaller routines?
- [ ] Can the readability be improved with different subroutine/variable names?
- [ ] Is the data flow understandable?
- [ ] Could comments in the code be improved to add clarity?
- [ ] Is there any commented out code?
- [ ] Does the code conform to the style guide (Fortran style compliance cannot be checked automatically)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -167,3 +177,133 @@ Once you have edited the script to your desired settings, run it.
...
+$ ../../exe/shallow_water_serialize.x ../../parm/shallow_water.nl
+```
+
+6) After installation of the Python Version requirements, execute the Python script to compare the Fortran Serialbox output with the GT4Py implementation. The `parm/shallow_water.yml` used in the script provides the same configuration as the `parm/shallow_water.nl` used to run the model on the Fortran side. 
+
+Validate against the Fortran implementation by running the following script:
+
+```bash
+$ cd src/python
+$ python shallow_water_serialize.py
```
Comment (christopherwharrop-noaa) [inline README.md]: Same for here:

```
[Christopher.W.Harrop@Hera:h33m42 python]$ mpiexec -n 1 python -m shallow_water_serialize ../../parm/shallow_water.yml
Finished running comparison tests, all PASSED!
[Christopher.W.Harrop@Hera:h33m42 python]$
```

**Review by christopherwharrop-noaa — CHANGES_REQUESTED** [review]
This update looks amazing!  Thank you so much for putting it all together.

Only a couple minor issues.  The instructions for using Serialbox to validate need to use the MPI launch commands instead of calling executable directly.  Also, due to the way the code works, we should update our existing namelists in `parm` so that the run steps is 735 instead of 734, just to make the logic for writing the serialization output save point and the output files a little easier and more clear.

**Comment by christopherwharrop-noaa** [comment]
I think the problem with the Intel Linux workflow seems to be with incorrect values for the compiler variables for the HDF5 install.  I think it should be:

```
          export CC=mpiicc
          export FC=mpiifort
          export CXX=mpiicpc
```

File: README.md (hunk 1 above)
Comment (robgpita) [reply README.md]: Thanks for the review, suggestions, and running the the Serialbox tests. I made the modifications you proposed above.

**Comment by christopherwharrop-noaa** [comment]
@robgonzalezpita I think we may need to update the URL for the OneAPI install for linux intel.

**Comment by robgpita** [comment]
@christopherwharrop-noaa Looks like those last couple of commits were what we needed (Updating the URL for the Intel HPC Toolkit for Linux , and removing the `--continue-with-optional-error` option).

**Review by christopherwharrop-noaa — APPROVED** [review]
Thank you so much for your work here.
